blackboxopt/optimization_loops/sequential.py

```diff
@@ -76,7 +77,10 @@ def run_optimization_loop(
 
         try:
             evaluation_specification = optimizer.generate_evaluation_specification()
-            logger.debug(f"Evaluating: {evaluation_specification}")
+            logger.info(
+                "The optimizer proposed a specification for evaluation:\n"
+                + f"{json.dumps(evaluation_specification.to_dict(), indent=2)}"
+            )
```

Comment:
Intuitively I would have use [`pprint.pformat()`](https://docs.python.org/3/library/pprint.html#pprint.pformat) instead of `json.dumps()`. But I guess there isn't much difference, beside outputting json vs. dict-format.

Reply:
I wonder would it prettify ` "configuration_key": [
      8,
      0,
      0
    ],` 

to make it a one-liner?

Reply:
I wonder if `pprint.pformat` will make a list to one-liner... That is the only thing that i don't like about current output (e.g. `configuration_key`)

I've just tested it and pprint makes it more compact, which is nice.
Howeve, the order has changed and quotes has changes, and types like boolean and `None` are not true json anymore. If somebody would want to copy and paste directly back this json to our server, there will be an error i assume:
Json:
````
The optimizer proposed a specification for evaluation:
{
  "configuration": {
    "x": 0.7652686018123811
  },
  "settings": {
    "fidelity": 1.0
  },
  "optimizer_info": {
    "model_based_pick": true,
    "configuration_key": [
      8,
      0,
      0
    ],
    "id": "c0fa7406-7bca-41a5-80aa-5bbf5c65ac8c"
  },
  "created_unixtime": 1642083992.064288,
  "context": null
}
````
Pprint:
````
The optimizer proposed a specification for evaluation:
{ 'configuration': {'x': 0.7652686018123811},
  'context': None,
  'created_unixtime': 1642083992.064288,
  'optimizer_info': { 'configuration_key': [8, 0, 0],
                      'id': 'c0fa7406-7bca-41a5-80aa-5bbf5c65ac8c',
                      'model_based_pick': True},
  'settings': {'fidelity': 1.0}}
```

Reply:
Given that there seems to be no trivial way to get the compactness of pprint with a JSON compatible output, let's go with json.dumps